pkg/orchestrator/provider/kubernetes/provider.go

```diff
@@ -0,0 +1,87 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/openclarity/vmclarity/api/models"
+	"github.com/openclarity/vmclarity/pkg/orchestrator/provider"
+)
+
+type Provider struct {
+	clientset kubernetes.Interface
+	config    *Config
+}
+
+var _ provider.Provider = &Provider{}
+
+func New(ctx context.Context) (provider.Provider, error) {
+	config, err := NewConfig()
+	if err != nil {
+		return nil, fmt.Errorf("invalid configuration. Provider=%s: %w", models.Kubernetes, err)
+	}
+
+	var clientConfig *rest.Config
+	if config.KubeConfig == "" {
+		// If KubeConfig config option not set, assume we're running
+		// incluster.
+		clientConfig, err = rest.InClusterConfig()
+		if err != nil {
+			return nil, fmt.Errorf("unable to load client configuration: %w", err)
+		}
+	} else {
+		cc, err := clientcmd.LoadFromFile(config.KubeConfig)
+		if err != nil {
+			return nil, fmt.Errorf("unable to load client configuration: %w", err)
+		}
+		clientConfig, err = clientcmd.NewNonInteractiveClientConfig(*cc, "", &clientcmd.ConfigOverrides{}, nil).ClientConfig()
+		if err != nil {
+			return nil, fmt.Errorf("unable to load client configuration: %w", err)
```

Comment:
Maybe change this error message to something different than the two previous messages

Reply:
+1

Reply:
Updated please check the latest revision.